Forward WebContent logs to the UI process

pvollan opened the pull request:
#### 6584bb248cb9520297d961cad4f5b7f5aec35386
<pre>
Forward WebContent logs to the UI process
<a href="https://bugs.webkit.org/show_bug.cgi?id=278022">https://bugs.webkit.org/show_bug.cgi?id=278022</a>
<a href="https://rdar.apple.com/133759441">rdar://133759441</a>

Reviewed by Chris Dumez.

In order to address quarantine issues with forwarding to the Networking process, forward WebContent logs to the UI process instead.
This will attribute the WebContent logs to a large number of UI processes, instead of to the single Networking process.

* Source/WebKit/NetworkProcess/LogStream.cpp:
(WebKit::LogStream::logOnBehalfOfWebContent):
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp:
(WebKit::NetworkConnectionToWebProcess::setupLogStream): Deleted.
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.h:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.messages.in:
* Source/WebKit/Shared/LogStream.cpp: Renamed from Source/WebKit/NetworkProcess/LogStream.cpp.
(WebKit::LogStream::~LogStream):
(WebKit::LogStream::logOnBehalfOfWebContent):
(WebKit::LogStream::setup):
* Source/WebKit/Shared/LogStream.h: Renamed from Source/WebKit/NetworkProcess/LogStream.h.
* Source/WebKit/Shared/LogStream.messages.in: Renamed from Source/WebKit/NetworkProcess/LogStream.messages.in.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/UIProcess/Cocoa/WebProcessProxyCocoa.mm:
(WebKit::WebProcessProxy::setupLogStream):
(WebKit::WebProcessProxy::logOnBehalfOfWebContent): Deleted.
* Source/WebKit/UIProcess/WebProcessProxy.h:
* Source/WebKit/UIProcess/WebProcessProxy.messages.in:
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::ensureNetworkProcessConnection):
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::platformInitializeWebProcess):
(WebKit::WebProcess::setupLogStream):

Canonical link: <a href="https://commits.webkit.org/284411@main">https://commits.webkit.org/284411@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/02fd12b11a0c7e5072f22c958cc9a69b5db8e0bf

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/69300 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/48700 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/21973 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/73382 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/20458 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/56501 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/20308 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/55114 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/13578 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/72366 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/44435 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/59831 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/35593 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/41101 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/18834 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/63043 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/17606 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/75094 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/13282 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/16829 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/62781 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/13321 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/59914 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/62687 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/10704 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/4315 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10585 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/44504 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/45578 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/46773 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/45319 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

bnham commented on Source/WebKit/UIProcess/Cocoa/WebProcessProxyCocoa.mm:
```diff
@@ -259,5 +259,37 @@
 #endif
 
+#if ENABLE(LOGD_BLOCKING_IN_WEBCONTENT)
+void WebProcessProxy::logOnBehalfOfWebContent(std::span<const char> logSubsystemIncludingNullTerminator, std::span<const char> logCategoryIncludingNullTerminator, std::span<const char> logStringIncludingNullTerminator, uint8_t logType, int32_t pid)
```
If we are going to be logging from the UIProcess, then this should occur on a MessageReceiver that does not run on the main thread. Maybe it should even happen directly on the IPC receiver thread?

cdumez replied:
Has this been addressed? It is not obvious to me that this code is running on a background thread. If it is, we should add an assertion.

pvollan replied:
Good point! I added an assertion in the latest patch.

Thanks for reviewing!

cdumez commented on Source/WebKit/NetworkProcess/LogStream.cpp:
```diff
@@ -53,6 +53,8 @@ LogStream::~LogStream()
 
 void LogStream::logOnBehalfOfWebContent(std::span<const uint8_t> logSubsystem, std::span<const uint8_t> logCategory, std::span<const uint8_t> logString, uint8_t logType)
```
We need to move this code to `Shared/` or `UIProcess/`. Having it remain in `NetworkProcess/` doesn't make sense.

cdumez replied:
Let's move to `Shared/` with `LogStreamIdentifier.h`

pvollan replied:
Done :) Thanks for reviewing!

pvollan committed:
Forward WebContent logs to the UI process
https://bugs.webkit.org/show_bug.cgi?id=278022
rdar://133759441

Reviewed by Chris Dumez.

In order to address quarantine issues with forwarding to the Networking process, forward WebContent logs to the UI process instead.
This will attribute the WebContent logs to a large number of UI processes, instead of to the single Networking process.

* Source/WebKit/NetworkProcess/LogStream.cpp:
(WebKit::LogStream::logOnBehalfOfWebContent):
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp:
(WebKit::NetworkConnectionToWebProcess::setupLogStream): Deleted.
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.h:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.messages.in:
* Source/WebKit/Shared/LogStream.cpp: Renamed from Source/WebKit/NetworkProcess/LogStream.cpp.
(WebKit::LogStream::~LogStream):
(WebKit::LogStream::logOnBehalfOfWebContent):
(WebKit::LogStream::setup):
* Source/WebKit/Shared/LogStream.h: Renamed from Source/WebKit/NetworkProcess/LogStream.h.
* Source/WebKit/Shared/LogStream.messages.in: Renamed from Source/WebKit/NetworkProcess/LogStream.messages.in.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/UIProcess/Cocoa/WebProcessProxyCocoa.mm:
(WebKit::WebProcessProxy::setupLogStream):
(WebKit::WebProcessProxy::logOnBehalfOfWebContent): Deleted.
* Source/WebKit/UIProcess/WebProcessProxy.h:
* Source/WebKit/UIProcess/WebProcessProxy.messages.in:
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::ensureNetworkProcessConnection):
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::platformInitializeWebProcess):
(WebKit::WebProcess::setupLogStream):

Canonical link: https://commits.webkit.org/284411@main
